pkg/logical-cluster-manager/cluster-manager-controller.go

```diff
@@ -153,7 +153,13 @@ func (c *Controller) processAdd(ctx context.Context, key any) error {
 	newClusterConfig := key.(*v1alpha1apis.LogicalCluster)
 	clusterName := newClusterConfig.Spec.ClusterName
 
-	provider := clusterproviderclient.GetProviderClient(newClusterConfig.Spec.ProviderType, newClusterConfig.Spec.ProviderName)
+	providerInfo, err := c.clusterclientset.EdgeV1alpha1().ClusterProviderDescs().Get(ctx, newClusterConfig.Spec.ProviderName, v1.GetOptions{})
+	if err != nil {
+		logger.Error(err, "failed to get provider resource.")
+		return err
+	}
+
+	provider := clusterproviderclient.GetProviderClient(providerInfo.Spec.ProviderType, newClusterConfig.Spec.ProviderName)
```

Comment:
Please add some comment that this is temporary. We would need to path other parameters as well (e.g., the config in case of kcp provider)

Reply:
done